File: Leetcode/1431_Kids With the Greatest Number of Candies.c
#include <stdlib.h>
#include <stdbool.h>
bool* kidsWithCandies(int* candies, int candiesSize, int extraCandies, int* returnSize) {
    bool* result = (bool*)malloc(candiesSize * sizeof(bool));
    int maxCandy = candies[0];
    for (int i = 1; i < candiesSize; i++) {
        if (candies[i] > maxCandy) {
            maxCandy = candies[i];
        }
    }
    for (int i = 0; i < candiesSize; i++) {
        if (candies[i] + extraCandies >= maxCandy) {
            result[i] = true;
        } else {
            result[i] = false;
        }
    }
    *returnSize = candiesSize;
    return result;
}

File: 1431_Kids With the Greatest Number of Candies.py
class Solution:
    def kidsWithCandies(self, candies, extraCandies) :
        result = []
        highest_candy = max(candies)
        for i in range(len(candies)):            
            result.append(candies[i] + extraCandies >= highest_candy)
        return result
